tests/sgl/device/test_texture_loader.cpp

```
@@ -0,0 +1,64 @@
+    MemoryStream stream(data.data(), data.size());
+    TextureLoader loader{ref<Device>(ctx.device)};
+
+    ref<Texture> path_texture = loader.load_texture(path);
+    ref<Texture> stream_texture = loader.load_texture(&stream);
+
+    CHECK(stream_texture->type() == path_texture->type());
+    CHECK(stream_texture->format() == path_texture->format());
+    CHECK(stream_texture->width() == path_texture->width());
+    CHECK(stream_texture->height() == path_texture->height());
+    CHECK(stream_texture->mip_count() == path_texture->mip_count());
+    CHECK(stream_texture->array_length() == path_texture->array_length());
```

Cppcheck (2.21.0)

[style] The function 'capacity' is never used.

(unusedFunction)